user-impl/src/main/java/com/example/auction/user/impl/CredentialEntity.java

```diff
@@ -0,0 +1,45 @@
+package com.example.auction.user.impl;
+
+import com.lightbend.lagom.javadsl.persistence.PersistentEntity;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class CredentialEntity extends PersistentEntity<CredentialCommand, CredentialEvent, Optional<PCredential>> {
```

Comment:
I agree with having the logic in UserEntity. 

In past comments I simply stated `Credentials` is sometimes a more adequate name and that probably misslead @lakhina . Apologies for the confusion.